effekt/shared/src/main/scala/effekt/core/optimizer/StaticArguments.scala

```diff
@@ -120,22 +123,22 @@ object StaticArguments {
 
   def rewrite(s: Stmt)(using C: StaticArgumentsContext): Stmt = s match {
     case Stmt.Scope(definitions, body) =>
-      scope(rewrite(definitions), rewrite(body))
+      normal.Scope(rewrite(definitions), rewrite(body))
 
     case Stmt.App(b, targs, vargs, bargs) =>
       b match {
         // if arguments are static && recursive call: call worker with reduced arguments
         case BlockVar(id, annotatedTpe, annotatedCapt) if hasStatics(id) && within(id) =>
           val IsStatic(staticT, staticV, staticB) = C.statics(id)
-          app(C.workers(id),
+          Stmt.App(C.workers(id),
             dropStatic(staticT, targs),
             dropStatic(staticV, vargs).map(rewrite),
             dropStatic(staticB, bargs).map(rewrite))
-        case _ => app(rewrite(b), targs, vargs.map(rewrite), bargs.map(rewrite))
+        case _ => Stmt.App(rewrite(b), targs, vargs.map(rewrite), bargs.map(rewrite))
```

Comment:
Here we do not perform beta reduction as this will be done later by `Normalizer`.